lib/hls/manifest_text_parser.js

```diff
@@ -242,49 +247,71 @@ shaka.hls.ManifestTextParser = class {
       const parser = new shaka.util.TextParser(data);
       let blockAttrs;
 
-      // Regex: any number of non-equals-sign characters at the beginning
-      // terminated by comma or end of line
-      const valueRegex = /^([^,=]+)(?:,|$)/g;
-
-      const blockValue = parser.readRegex(valueRegex);
+      // Reset regex lastIndex to ensure proper matching
```

Comment:
Can we add a comment about the fact that we reset lastIndex because the regex contains a g(lobal) flag?